src/test_l1_dsFPD.c

```diff
@@ -1058,40 +1180,60 @@ void test_l1_dsFPD_negative_dsGetFPState (void)
  * |:--:|---------|----------|--------------|-----|
  * |01|Initialize with dsFPInit()||dsERR_NONE|Ensure the system is initialized|
  * |02|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
- * |03|Call dsSetFPColor() with all valid eIndicator and all valid eColor parameters from kIndidicators|eIndicator: [Valid Indicator], eColor: [Valid Color]|dsERR_NONE|Ensure that the function can set the color|
- * |04|Terminate with dsFPTerm()||dsERR_NONE|Ensure the system is terminated|
+ * |03|Call dsSetFPColor() with all valid eIndicator and all valid eColor parameters from the profile file |eIndicator: [Valid Indicator], eColor: [Valid Color]|dsERR_NONE|Ensure that the function can set the color|
+ * |04|Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_OFF |dsERR_OPERATION_NOT_SUPPORTED|Validate invalid parameter handling for eIndicator|
+ * |05|Terminate with dsFPTerm()||dsERR_NONE|Ensure the system is terminated|
  * 
- * @note Valid indicators can retrieved from id element in kIndicators in the dsFPDSettings.h file
+ * @note Valid indicators can retrieved from yaml file
  */
 void test_l1_dsFPD_positive_dsSetFPColor (void)
 {
     gTestID = 15;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     dsError_t result;
+    int count =0;
+    dsFPDColor_t color;
+    dsFPDIndicator_t eIndicator;
+    char buffer[DS_FPD_KEY_SIZE];
+    int numOfSupportedColors =0;
 
     // Step 01: Initialize with dsFPInit()
     result = dsFPInit();
     UT_ASSERT_EQUAL(result, dsERR_NONE);
 
     // Step 02: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
-    // Step 03: Call dsSetFPColor() with all valid eIndicator and all valid eColor parameters from kIndicators
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    // Step 03: Call dsSetFPColor() with all valid eIndicator and all valid eColor parameters from yaml
+    count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    for (int i = 0; i < count; i++)
     {
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);
+	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-        dsFPDColorConfig_t *colors = kIndicators[i].supportedColors;
+        numOfSupportedColors = UT_KVP_PROFILE_GET_LIST_COUNT("dsFPD/SupportedFPDIndicators/%d/supportedColors");
```

Comment:
what is %d here.  There is no indexing value or variable

Reply:
Done